Gobabygo/Gobabygo.ino

```diff
@@ -36,65 +36,63 @@ float correctForDeadzone(float axis, float deadzone) {
   }
 }
 
-
 // This funtion gradually accelerates a value toward a target
 float accelerateToValue(float currentVal, float targetVal, int timeElapsed, float acceleration) {
   return currentVal + (targetVal - currentVal) * timeElapsed * acceleration;
 }
 
-
 void setup() {
   // put your setup code here, to run once:
   Serial.begin(9600);
   pinMode(rightMotorPin, OUTPUT);
   pinMode(leftMotorPin, OUTPUT);
 }
 
-
 void loop() {
   // put your main code here, to run repeatedly:
   currentTime = millis();
   timeSincePrint += currentTime - previousTime;
   timeSinceAcceleration = currentTime - previousTime;
   previousTime = currentTime;
   
+  float rawJoyXValue = analogRead(joyStickXPin);
   joyXVal = analogRead(joyStickXPin);
-  joyXVal = 2/1023 * joyXVal - 1;
+  joyXVal = 2.0/1023.0 * joyXVal - 1;
   joyXVal = correctForDeadzone(joyXVal, deadzone);
   joyXVal = accelerateToValue(joyXVal, previousJoyXVal, timeSinceAcceleration, accelerationPerMs);
   previousJoyXVal = joyXVal;
   
+  float rawJoyYValue = analogRead(joyStickYPin);
   joyYVal = analogRead(joyStickYPin);
-  joyYVal = 2/1023 * joyYVal -1;
+  joyYVal = 2.0/1023.0 * joyYVal -1;
   joyYVal = correctForDeadzone(joyYVal, deadzone);
   joyYVal = accelerateToValue(joyYVal, previousJoyYVal, timeSinceAcceleration, accelerationPerMs);
   previousJoyYVal = joyYVal;
   
   if (joyYVal < 0) {
     joyYVal *= backwardPower;
   }
-  else {
-    joyXVal -= joyYVal * turnScaler;
-  }
-
+ 
   // Cheesy drive logic
   leftMotorVal = joyYVal + joyXVal;
   rightMotorVal = joyYVal - joyXVal;
-
+  
   if (abs(leftMotorVal) > 1 or abs(rightMotorVal) > 1) {  // the value of either motor can never be outside the range -1 to 1
     scaler = max(abs(leftMotorVal), abs(rightMotorVal));
     leftMotorVal = leftMotorVal / scaler;
     rightMotorVal = rightMotorVal / scaler;
   }
   
-  analogWrite(leftMotorPin, leftMotorVal * 127.5 + 127.5);    // converting from -1-1 to 0-255
-  analogWrite(rightMotorPin, rightMotorVal * 127.5 + 127.5);
+  //analogWrite(leftMotorPin, leftMotorVal * 127.5 + 127.5);    // converting from -1-1 to 0-255
+  //analogWrite(rightMotorPin, rightMotorVal * 127.5 + 127.5);
```

Comment:
You could leave these two lines active in your official code ... we just disabled today since Zach was seeing strange motor controller behavior